### PR TITLE
Confusion matrix normalized display

### LINKED ISSUE #1115: [FEAT] Normalize confusion matrix by class support size
**Is your feature request related to a problem? Please describe.**
Similarly to [sklearn](https://scikit-learn.org/stable/auto_examples/model_selection/plot_confusion_matrix.html), there should be an option to show the confusion matrix normalized by class support size, such that the quantity in each matrix cell is n_in_cell / number_of_samples_with_this_label. 
@shir22 IMO this option should be true by default, which is not the way it is in sklearn. 

**Describe the solution you'd like**
In case the flag is true, the display values will be the fractions, and the count (number of samples in cell) will be displayed in the hover values. In case the flag is false, the display value will be the count, the the fraction would be the hover value.

**Bonus points**
Also enable normalizing by the number of samples predicted to this class, rather than normalizing by the number of samples actually belonging to this class.

### PR DESCRIPTION
**Pull request by JKL98ISR**
closes #1115
tests just checking if it works

### CODE REVIEW

File: deepchecks/tabular/checks/performance/confusion_matrix_report.py [hunk 1]
```diff
@@ -11,17 +11,30 @@
...
 
 
 class ConfusionMatrixReport(SingleDatasetCheck):
-    """Calculate the confusion matrix of the model on the given dataset."""
+    """Calculate the confusion matrix of the model on the given dataset.
+
+    Parameters
+    ----------
+    is_normalize (bool, default True):
+        If to color the matrix by normalizing by true values.
```
Comment (matanper) [inline deepchecks/tabular/checks/performance/confusion_matrix_report.py:30]: this sentence is not clear

File: deepchecks/vision/checks/performance/confusion_matrix.py [hunk 2]
```diff
@@ -143,18 +148,11 @@ def compute(self, context: Context, dataset_kind: DatasetKind = None) -> CheckRe
                 x.append(it)
                 y.append(it)
             else:
-                x.append('No overlapping prediction')
-                y.append('No overlapping label')
+                x.append('No overlapping')
```
Comment (matanper) [inline deepchecks/vision/checks/performance/confusion_matrix.py:151]: why removed prediction/label?

Comment (matanper) [inline deepchecks/vision/checks/performance/confusion_matrix.py:151]: why removed prediction/label?

File: deepchecks/tabular/checks/performance/confusion_matrix_report.py (hunk 1 above)
Comment (ItayGabbay) [reply deepchecks/tabular/checks/performance/confusion_matrix_report.py:29]: agree

File: deepchecks/utils/plot.py [hunk 3]
```diff
@@ -158,3 +169,51 @@ def shifted_color_map(cmap, start=0, midpoint=0.5, stop=1.0, name: str = 'shifte
...
+
+    fig = go.Figure(data=go.Heatmap(
+                x=x,
+                y=y,
+                z=confusion_matrix_norm,
+                text=confusion_matrix,
+                texttemplate=texttemplate))
+    fig.data[0].colorbar.title = colorbar_title
+    fig.update_layout(title=plot_title)
+    fig.update_layout(width=600, height=600)
```
Comment (ItayGabbay) [inline deepchecks/utils/plot.py:215]: don't we want to remove all fixed sizes from plots?

File: deepchecks/vision/checks/performance/confusion_matrix.py (hunk 2 above)
Comment (ItayGabbay) [reply deepchecks/vision/checks/performance/confusion_matrix.py:151]: agree

Comment (JKL98ISR) [reply deepchecks/vision/checks/performance/confusion_matrix.py:151]: it makes the confusion matrix much smaller

File: deepchecks/tabular/checks/performance/confusion_matrix_report.py (hunk 1 above)
Comment (JKL98ISR) [reply deepchecks/tabular/checks/performance/confusion_matrix_report.py:29]: can you give me a better option?

File: deepchecks/utils/plot.py (hunk 3 above)
Comment (JKL98ISR) [reply deepchecks/utils/plot.py:215]: it looks bad without it

**Comment by JKL98ISR** [comment]
Screen Shot 2022-05-03 at 16 18 51

File: deepchecks/tabular/checks/performance/confusion_matrix_report.py (hunk 1 above)
Comment (nirhutnik) [reply deepchecks/tabular/checks/performance/confusion_matrix_report.py:29]: "boolean that determines whether to normalize the true values of the matrix"?
Also, maybe "should_normalize" or just "normalize"? I dislike "is_..." format.

File: deepchecks/utils/plot.py [hunk 4]
```diff
@@ -11,10 +11,21 @@
...
+from deepchecks.utils.strings import format_number
+
+__all__ = ['create_colorbar_barchart_for_check', 'shifted_color_map',
+           'create_confusion_matrix_figure', 'colors', 'hex_to_rgba']
+
+
+def _format_number_if_not_nan(x):
+    if np.isnan(x):
+        return x
+    return format_number(x)
```
Comment (nirhutnik) [inline deepchecks/utils/plot.py:27]: Why not put this in string_utils, together with format_number?

File: deepchecks/utils/plot.py [hunk 5]
```diff
@@ -158,3 +169,51 @@ def shifted_color_map(cmap, start=0, midpoint=0.5, stop=1.0, name: str = 'shifte
 def hex_to_rgba(h, alpha):
     """Convert color value in hex format to rgba format with alpha transparency."""
     return 'rgba' + str(tuple([int(h.lstrip('#')[i:i+2], 16) for i in (0, 2, 4)] + [alpha]))
+
+
+def create_confusion_matrix_figure(confusion_matrix, x, y, is_normalize):
```
Comment (nirhutnik) [inline deepchecks/utils/plot.py:174]: Fill in types of inputs in signature as well

File: deepchecks/utils/plot.py [hunk 6]
```diff
@@ -158,3 +169,51 @@ def shifted_color_map(cmap, start=0, midpoint=0.5, stop=1.0, name: str = 'shifte
...
+    """Create a confusion matrix figure.
+
+    Parameters
+    ----------
+    confusion_matrix: np.ndarray
+        array containing the confusion matrix.
```
Comment (nirhutnik) [inline deepchecks/utils/plot.py:180]:
```suggestion
        2D array containing the confusion matrix.
```

File: deepchecks/utils/plot.py [hunk 7]
```diff
@@ -158,3 +169,51 @@ def shifted_color_map(cmap, start=0, midpoint=0.5, stop=1.0, name: str = 'shifte
...
+    x: np.ndarray
+        array containing x axis data.
+    y: np.ndarray
+        array containing y axis data.
+    is_normalize: bool
+        if True will also show normalized values by the true values.
+
+    Returns
+    -------
+    go.Figure
```
Comment (nirhutnik) [inline deepchecks/utils/plot.py:190]:
```suggestion
    plotly Figure object
```

File: deepchecks/utils/plot.py [hunk 8]
```diff
@@ -158,3 +169,51 @@ def shifted_color_map(cmap, start=0, midpoint=0.5, stop=1.0, name: str = 'shifte
...
+        confusion matrix figure
+
+    """
+    if is_normalize:
```
Comment (nirhutnik) [inline deepchecks/utils/plot.py:194]: If the values are between 0 and 100 it's not commonly referred to as "normalization". Though I guess Bressler asked for it to be in percentages and not in [0,1] range?
I don't have any other suggestion though.

File: deepchecks/utils/plot.py [hunk 9]
```diff
@@ -158,3 +169,51 @@ def shifted_color_map(cmap, start=0, midpoint=0.5, stop=1.0, name: str = 'shifte
...
+        confusion_matrix_norm = confusion_matrix.astype('float') / \
+            confusion_matrix.sum(axis=1)[:, np.newaxis] * 100
+        confusion_matrix_norm = np.vectorize(_format_number_if_not_nan)(confusion_matrix_norm)
+        texttemplate = '%{z}%<br>(%{text})'
+        colorbar_title = '% of<br>True Values'
+        plot_title = 'Percent Out of True Values (Count)'
+    else:
+        confusion_matrix_norm = confusion_matrix
```
Comment (nirhutnik) [inline deepchecks/utils/plot.py:202]: Why call it norm if it's not normalized?

File: deepchecks/vision/checks/performance/confusion_matrix.py [hunk 10]
```diff
@@ -59,17 +59,22 @@ class ConfusionMatrixReport(SingleDatasetCheck):
         Threshold to consider bounding box as detected.
     iou_threshold (float, default 0.5):
         Threshold to consider detected bounding box as labeled bounding box.
+    is_normalize (bool, default True):
+        If to color the matrix by normalizing by true values.
```
Comment (nirhutnik) [inline deepchecks/vision/checks/performance/confusion_matrix.py:63]: Notice the same comment on phrasing is relevant here

**Review by nirhutnik — APPROVED** [review]

**Review by matanper — APPROVED** [review]

**Review by ItayGabbay — APPROVED** [review]
